src/memory/combat_manager.rs

```diff
@@ -2,12 +2,95 @@ use crate::memory::memory_context::MemoryContext;
 use crate::memory::{MemoryManager, MemoryManagerUpdate};
 use crate::state::StateContext;
 use log::info;
+use memory::game_engine::il2cpp::unity_list::{UnityItem, UnityList};
 use memory::memory_manager::il2cpp::UnityMemoryManager;
 use memory::process::MemoryError;
+use memory::process::Process;
+use memory::string::*;
+
+#[derive(Default, Debug)]
+pub enum CombatControllerType {
+    #[default]
+    Basic,
+    FirstEncounter,
+    SecondEncounter,
+    DwellerOfStrife,
+    DwellerOfDread,
+    KOTutorial,
+    LiveManaTutorial,
+    ManaRegenTutorial,
+    RoundsTutorial,
+    SpellLockTutorial,
+    TimedBlocksTutorial,
+    TimedHitsTutorial,
+}
+
+#[derive(Debug, Default, Clone)]
+pub enum CombatDamageType {
```

Comment:
Maybe move the things that are not `CombatManagerData` into own file? Not sure if any of them belong in the `data` crate, but it could be a place for them (like with `PlayerPartyCharacter`).

Reply:
Tried the data crate, but these objects are used in UnityList; Unfortunately rust doesn't let you impl Traits from other crates: https://github.com/Ixrec/rust-orphan-rules?tab=readme-ov-file#what-are-the-orphan-rules